dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ExecutionModel.java

```diff
@@ -464,11 +467,10 @@ private void extractReadsFrom() {
             for (EventData read : addressedReads.getValue()) {
                 for (EventData write : addressWritesMap.get(address)) {
                     BooleanFormula rfExpr = rf.encode(write.getEvent(), read.getEvent());
-                    // The null check is important: Currently there are cases where no rf-edge between
+                    // The null check in isTrue is important: Currently there are cases where no rf-edge between
                     // init writes and loads get encoded (in case of arrays/structs). This is usually no problem,
                     // since in a well-initialized program, the init write should not be readable anyway.
-					Boolean rfVal = model.evaluate(rfExpr);
-					if (rfVal != null && rfVal) {
+					if (isTrue(rfExpr)) {
```

Comment:
Isn't the comment outdated? It seems you are removing the null check

Reply:
The null-check is implicit: `TRUE.equals(null) == false`.